controlplane/src/core/repositories/FederatedGraphRepository.ts

```diff
@@ -162,21 +164,25 @@ export class FederatedGraphRepository {
         await targetRepo.updateReadmeOfTarget({ id: data.targetId, readme: data.readme });
       }
 
-      if (labelMatchers.length > 0) {
-        // update label matchers
+      // update label matchers
+      if (data.labelMatchers.length > 0 || data.unsetLabelMatchers) {
+        const labelMatchers = normalizeLabelMatchers(data.unsetLabelMatchers ? [] : data.labelMatchers);
+
         await tx
           .delete(schema.targetLabelMatchers)
           .where(eq(schema.targetLabelMatchers.targetId, federatedGraph.targetId));
 
-        await tx
-          .insert(schema.targetLabelMatchers)
-          .values(
-            labelMatchers.map((labelMatcher) => ({
-              targetId: federatedGraph.targetId,
-              labelMatcher: labelMatcher.split(','),
-            })),
-          )
-          .execute();
+        if (labelMatchers.length > 0) {
+          await tx
+            .insert(schema.targetLabelMatchers)
+            .values(
+              labelMatchers.map((labelMatcher) => ({
```

Comment:
Don't we need to call `normalizeLabelMatchers` here?

Reply:
It is called